worklist/models.py

```diff
@@ -2,26 +2,88 @@
 from django.db import models
 
 
-# Model for table that contains the data
+# Model for table that contains the data for Worklists
 class WorkList(models.Model):
-    id = models.AutoField(primary_key=True)
-    name = models.CharField(max_length=100, blank=True)
-    theme = models.CharField(max_length=100, blank=True)
-    description = models.CharField(max_length=100, blank=True)
-    created_by = models.CharField(max_length=100, blank=True)
+    worklist_id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=100)
+    tags = models.CharField(max_length=200, blank=True)
+    description = models.CharField(max_length=300, blank=True)
+    created_by = models.CharField(max_length=100)
     date_created = models.DateTimeField(auto_now_add=True)
     date_updated = models.DateTimeField(auto_now=True)
     psid = models.IntegerField(blank=True)
-    psid_added_by = models.CharField(max_length=100, blank=True)
+
+    class Meta:
+        unique_together = ('name', 'created_by')
 
     # Method to create a new entry in the table
     @staticmethod
     def create_object(data):
-        WorkList.objects.create(name=data['name'],
-                                theme=data['theme'],
-                                description=data['description'],
-                                created_by=data['created_by'],
-                                psid=data['psid'],
-                                psid_added_by=data['psid_added_by'])
+        obj, created = WorkList.objects.get_or_create(name=data['name'],
+                                                      tags=data.get('tags', ''),
+                                                      description=data.get('description', ''),
+                                                      created_by=data['created_by'],
+                                                      psid=data.get('psid', 0))
+        return obj
+
 
+# Model for table that contains the data for Articles
+class Articles(models.Model):
+    name = models.CharField(primary_key=True, max_length=100)
+    article_id = models.IntegerField(blank=True, null=True)
+    avg_page_views = models.IntegerField(blank=True)
+    projects = models.CharField(max_length=300, blank=True)
+    size = models.IntegerField(blank=True)
+    grade = models.CharField(max_length=300, blank=True)
 
+    # Method to create a new entry in the table
+    @staticmethod
+    def create_object(data):
+        if data.get('article_id') is None:
+            data['article_id'] = 0
+
+        obj, created = Articles.objects.get_or_create(article_id=data.get('article_id', 0),
```

Comment:
Don't know if it's just GitHub formatting, but indentation looks very off here. Maybe the line is too long?

Reply:
It's because of GitHub's formatting only. In actual it is properly indented.